vyper/parser/arg_clamps.py

```diff
@@ -64,10 +65,28 @@ def make_arg_clamper(datapos, mempos, typ, is_init=False):
         ], typ=None, annotation='checking bytearray input')
     # Lists: recurse
     elif isinstance(typ, ListType):
-        o = []
-        for i in range(typ.count):
-            offset = get_size_of_type(typ.subtype) * 32 * i
-            o.append(make_arg_clamper(datapos + offset, mempos + offset, typ.subtype, is_init))
+        if typ.count > 5 or (type(datapos) is list and type(mempos) is list):
```

Comment:
when are `datapos` and `mempos` lists?

Reply:
When we have a list of lists!

Reply:
which test case?

Reply:
@siraben I think you should jsut use get_size_of_type(typ) here instead this will support above cases as well.